server/plugin/api.go

```diff
@@ -984,10 +987,29 @@ func (p *Plugin) getYourAssignments(c *UserContext, w http.ResponseWriter, r *ht
 	result, _, err := githubClient.Search.Issues(c.Ctx, query, &github.SearchOptions{})
 	if err != nil {
 		c.Log.WithError(err).With(logger.LogContext{"query": query}).Warnf("Failed to search for assignments")
-		return
+		return nil
 	}
 
-	p.writeJSON(w, result.Issues)
+	return result.Issues
+}
+
+func (p *Plugin) getSidebarData(c *UserContext) *SidebarContent {
+	assignments := p.getYourAssignmentsData(c)
+	prs := p.getYourPrsData(c)
+	reviews := p.getReviewsData(c)
+	unreads := p.getUnreadsData(c)
```

Comment:
If these functions are not APIs I think we can move them to utils folder if there is one?

Reply:
These are not pluginAPIs anymore but still they are calling APIs of github to get the data so I think we can keep it here.